llvm/lib/Target/AIE/AIEPostSelectOptimize.cpp

```diff
@@ -266,6 +276,90 @@ bool combineINSERT_SUBREG(MachineBasicBlock &MBB) {
   return Changed;
 }
 
+/// This optimization prevents the coalescing and copy propagation of
+/// Addressing registers for small loops. This is important to avoid
+//  unnecessary COPYs of such descriptors when they are coalesced.
+bool duplicateAdressingRegs(MachineBasicBlock &MBB, MachineRegisterInfo &MRI) {
+
+  const TargetInstrInfo &TII = *MBB.getParent()->getSubtarget().getInstrInfo();
+  bool Changed = false;
+
+  // TODO: LoopUtils?
```

Comment:
We can postpone this refactoring for now to avoid conflicts.

Reply:
Nit: Now that https://github.com/Xilinx/llvm-aie/pull/152 is merged, we might be able to share the code already. Not sure if there will be more conflicts with https://github.com/Xilinx/llvm-aie/pull/146

Reply:
We now have `llvm/lib/Target/AIE/Utils/AIELoopUtils.h`. Maybe we should move the helper there?

Reply:
I moved this lambda as a function to AIELoopUtils.